src/metrics_server.rs

```diff
@@ -0,0 +1,171 @@
+
+            let time_since_the_epoch = SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .expect("Time went backwards");
+            let time_elapsed = time_since_the_epoch
+                .checked_sub(Duration::from_secs(latest_block_timestamp as u64))
+                .unwrap();
+
+            if latest_block_timestamp != 0.0 {
+                match time_elapsed.as_secs().ge(&Duration::new(150, 0).as_secs()) {
+                    true => probe_result = ProbeResult::SomethingIsWrong,
+                    false => probe_result = ProbeResult::IndexerStarted,
+                }
+            }
+            let mut res = "".to_owned();
+
+            match probe_result {
+                ProbeResult::IndexerNotStarted => res.push_str("Indexer has not started yet."),
+                ProbeResult::IndexerStarted => {
+                    res.push_str("Indexer is operating normally. Blocks reported at least every 2.5 minutes.");
+                }
+                ProbeResult::SomethingIsWrong => {
+                    res.push_str("Something is wrong. Indexer has not reported any new blocks within 2.5 minutes.");
+                }
+            };
+
+            if latest_block_timestamp != 0.0 {
+                res.push_str("\n Last block height was: ");
+                res.push_str(last_known_blockheight.to_string().as_str());
+                res.push_str("\n last block was indexed at timestamp: ");
+                let last_block_datetime =
+                    NaiveDateTime::from_timestamp_opt(latest_block_timestamp as i64, 0);
+
+                match last_block_datetime {
+                    Some(datetime) => {
+                        let dt = DateTime::<Utc>::from_utc(datetime, Utc);
+                        res.push_str(&dt.format("%a %b %e %T %Y").to_string());
+                    }
+                    None => res.push_str("Error Converting time."),
+                }
+            }
+
+            Response::builder()
+                .status(200)
+                .header(CONTENT_TYPE, encoder.format_type())
+                .body(Body::from(res))
+                .unwrap()
+        }
+        (&Method::GET, "/") => Response::builder()
+            .status(200)
+            .body(Body::from("Service is running."))
+            .unwrap(),
+        _ => Response::builder()
+            .status(404)
+            .body(Body::from("Missing Page"))
+            .unwrap(),
+    };
+
+    Ok(response)
+}
+
+#[tokio::main]
+pub async fn init_metrics_server() -> std::result::Result<(), ()> {
+    // For every connection, we must make a `Service` to handle all
+    // incoming HTTP requests on said connection.
+    let make_svc = make_service_fn(move |_conn| async move {
+        Ok::<_, Infallible>(service_fn(serve_req))
+    });
+
+    let port: u16 = match env::var("PORT") {
+        Ok(val) => val.parse::<u16>().unwrap(),
+        _ => 3000,
```

Comment:
it should be the other env variable

Reply:
Cloud run by default listens to $PORT to figure out if the service is running or not.